include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -64,16 +65,21 @@ inline bool search_trivial(cursor_t cur, query_t & query, typename cursor_t::siz
     // Approximate case
     else
     {
+        // Do not use insertion if there is a match.
+        // Check if cursor did a step in the index.
+        bool const do_insertion = cur.query_length() > 0 ? cur.last_char() != query[query_pos] : true;
+        
         // Insertion
-        if (error_left.insertion > 0)
+        // Do not allow insertions after an deletion.
```

Comment:
I think if you do not allow any substitutions, you need to allow insertions after deletions:
```
text1
GGCCTTTA
text2
GGTCTGTT
query
GGCCTGTT
```
Allowing 2 errors, 0 sub, 1 del, 1 ins:
```
GGCCT-TTA
GG-CTGTT
```
But also
```
GGT-CTGTT
GG-CCTGTT
```

Reply:
You are right, I will also look into doing a unit test for this.